### DISCUSSION TITLE
pybricksdev run without file argument to run program already in hub

### DISCUSSION THREAD

**Discussion by vascolp** [pybricks/support#1159] [Ideas]
Can you do a pybricksdev run like command but without the <file> argument to run the program already in the hub?
Could be as simple as not having the <file> argument in the run command.
So that we can see the results of the program in the output
With the storage data, we can modify program brehaviour without the need of always loading a new program, this would be useful...

Just a small feature, but thanks in advance!

**Comment by dlech** [comment]
There isn't a command line option for this currently, but you could write your own program that connects to a hub and calls `start_user_program()`.

**Comment by vascolp** [comment]
I  guess I can do that yes... but then I would have to take care of all hub output, which pybricksdev does very well.
And it seems so simple to do it in pybricksdev run... I mean it looks like it is only not to execute the program upload part...
Maybe when you have time...

**Reply by Novakasa** [reply]
I think this was implemented for the `PybricksHub.run` method in https://github.com/pybricks/pybricksdev/pull/53 . Not as a command line option though. But if you run a python script, you can use the `PybricksHub.run` method and don't provide a `path` parameter.

The script might look something like this:
```python
import asyncio
from pybricksdev.ble import find_device
from pybricksdev.connections.pybricks import PybricksHub

async def main():
    dev = await find_device()
    hub = PybricksHub()
    await hub.connect(dev)
    await hub.run(print_output=True, wait=True)
    # await hub.run("hub_program.py", print_output=True, wait=True)   # This would download the program "hub_program.py"
    await hub.disconnect()

asyncio.run(main())
 ```
